Fedora flavors

fepitre opened the pull request:
Add Fedora desktop flavors: here is a first one I'm using since few months XFCE. Uniform look with dom0 and lighter image.

Also, I have made a script for custom Thunar actions (copy, move, convert-img, convert-pdf, openvm and opendvm), for which I will do a separate package or we can integrate it with other utils.

marmarek commented on template_scripts/04_install_qubes.sh:
```diff
@@ -63,6 +63,16 @@ if [ "0$TEMPLATE_ROOT_WITH_PARTITIONS" -eq 1 ]; then
+    case "$TEMPLATE_FLAVOR" in
+      "xfce")
+          echo "--> Setting up XFCE xinitrc script..."
+          cp ${SCRIPTSDIR}/extras/50-xfce-desktop.sh ${INSTALLDIR}/etc/X11/xinit/xinitrc.d
```
Better package such stuff somewhere, instead of copying during template build. Otherwise there is no way to update such file in a sane way. I think this belongs to gui-agent-linux/appvm-scripts/etc/X11/xinit/xinitrc.d. Can be either packaged as a separate one-file sub-package, or enabled with some configuration.

fepitre replied:
Thank you for your feedback. This is what I thought about but before modifying something in other packages I wanted some feedback from someone else. I will prepare it.

marmarek commented:
Also, xfce flavor fails to build, see here: https://travis-ci.org/marmarek/qubes-builder-fedora/builds/297043866

marmarek commented:
Is it meant only for Fedora 26+?

fepitre commented:
It is meant for Fedora 25+, this is when I started using it. I also built it for Fedora 26 yesterday and I indeed I had not the conflict with firewalld (for which I had once for Fedora 25 but I though it was solved). I will launch the build for Fedora 25 and investigate.

Also, a remark with respect to packages list. Like CentOS 7, for Fedora 27 I had to replace the workstation package by gnome-desktop because it includes so much groups dependencies who failed due to excluded package like avahi etc. I update things for FC27 in the next days and I will show you.

fepitre commented:
I put things related to xinit into gui-agent-linux (https://github.com/fepitre/qubes-gui-agent-linux/commit/53c2781d3bfae2925c5a6bb5738a45653b51b859). I also add Thunar actions. It generates gui-agent-xfce rpm. In consequence, I added in 04_install_qubes script a flavor selection with respect to a qubes packages list. I created packages_qubes_xfce.list from packages_qubes.list where I added gui-agent-xfce. The template is built and I will test it tonight or tomorrow (i'm not currently at home).

fepitre commented:
I reopen it in another branch.
